web-portal/backend/src/usage/usage.service.ts

```diff
@@ -0,0 +1,85 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { createHash } from 'crypto';
+
+@Injectable()
+export class UsageService {
+  async getAppUsage(appId: string, period: string): Promise<any> {
+    const hashedAppId = createHash('sha256').update(appId).digest('hex');
+    const stepBack = this.getSeconds(period);
+    const step = this.getStep(period);
+    if (!stepBack || !step) {
+      throw new HttpException('Invalid period', HttpStatus.BAD_REQUEST);
+    }
+
+    const start = new Date().getTime() - stepBack;
+    const end = new Date().getTime();
+    const q = `gateway_relay_usage{appId="${hashedAppId}"}`;
+
+    console.log({step, stepBack, start, end, q})
+    const result = await this.fetchData(q, start, end, step);
+    return result.json();
+  }
+
+  async getTenantUsage(tenantId: string, period: string): Promise<any> {
+    const stepBack = this.getSeconds(period);
+    const step = this.getStep(period);
+    if (!stepBack || !step) {
+      throw new HttpException('Invalid period', HttpStatus.BAD_REQUEST);
+    }
+
+    const start = new Date().getTime() - stepBack;
+    const end = new Date().getTime();
+    const q = `gateway_relay_usage{tenant="${tenantId}"}`;
+
+    console.log({step, stepBack, start, end, q})
+
+    const result = await this.fetchData(q, start, end, step);
+    return result.json();
+  }
+
+  private async fetchData(query: string, start: number, end: number, step: number | string): Promise<Response> {
+    const url = `https://api.fly.io/prometheus/porters-staging/api/v1/query_range?query=${query}&start=${start}&end=${end}&step=${step}`;
```

Comment:
Definitely want to move this to env var, hardcoding staging is something that will sneaky bite when moving to prod.